tests/lean/interactive/goTo.lean

```diff
@@ -61,6 +61,53 @@ macro_rules | `(test) => `(3)
 #check test
      --^ textDocument/definition
 
+class Foo2 where
+  foo : Nat → Nat
+  foo' : Nat
+
+class Foo3 [Foo2] where
+  foo : [Foo2] → Nat
+
+class inductive Foo4 : Nat → Type where
+| mk : Nat → Foo4 0
+
+def Foo4.foo : [Foo4 n] → Nat
+| .mk n => n
+
+class Foo5 where
+  foo : Foo2
+
+
+instance : Foo2 := .mk id 0
+instance : Foo3 := .mk 0
+instance : Foo4 0 := .mk 0
+instance [foo2 : Foo2] : Foo5 := .mk foo2
+
+-- should go-to instance
+              --v textDocument/definition
+#check Foo2.foo  2
+          --^ textDocument/definition
+#check Foo2.foo
```

Comment:
Reporting back after a merge with master -- It seems that this syntax is now elaborated to have the type `[self : Foo2] → Nat → Nat`, which is why the tests are failing now. Did something change recently for the `self` TC parameter to be now be interpreted as strict implicit?

Reply:
That is https://github.com/leanprover/lean4/pull/1943